### DISCUSSION TITLE
sha256pass not working as expected on sqlauth

### DISCUSSION THREAD

**Discussion by ghost** [inspircd/inspircd#1828] [Support]
**Description**
I try to use the sha256 module with sqlauth but when I login it returns "SQL query returned no matches" though the credentials are matching.

The SHA256 was generated with python3.9 (like so: hashlib.sha256('example'.encode('utf-8')).hexdigest())

For example: If i do: echo 'password' | sha256sum I get a non-matching result.
But when I do: echo -n 'password' | sha256sum, it matches.

This error is caused in the way the sha256 module generates the hash?

When I remove $nick from the query , it fails, when I remove $sha256pass from the query, it works. So it's definitely the hash that doesn't match.

**Steps to reproduce the issue:**

1. generate hash with pyhton (hexdigest) and store it in a db.
2. configure inspircd with sqlauth with sha256pass
3. login to irc with creds

**Describe the results you received:**
Not logged in with error: SQL query returned no matches

**Describe the results you expected:**
Logged in with message: CONNECT

**Additional information you deem important (e.g. issue happens only occasionally):**
Installed with apt on Debian 10.

**Output of `./bin/inspircd --version`:**
InspIRCd-2.0.27 r0

**Comment by ghost** [comment]
doing some more research, this depends on the client used. ex.: With Hexchat the auth works fine, but Kiwiirc fails to auth.

**Comment by sadiepowell** [comment]
>For example: If i do: echo 'password' | sha256sum I get a non-matching result.

>But when I do: echo -n 'password' | sha256sum, it matches.

Yes, that's working as intended. Without the `-n` echo option sha256sum will hash the terminating newline as part of the password. See `man echo`.

>With Hexchat the auth works fine, but Kiwiirc fails to auth.

If it's only working in some clients then thats probably an issue with how those clients have been configured.

---

Additionally, v2 goes EOL in under a week so you should really upgrade, especially if you're using sqlauth because of security advisories [2019-02](https://docs.inspircd.org/security/2019-02/) and [2020-01](https://docs.inspircd.org/security/2020-01/). There are .deb files for Debian and Ubuntu on [the downloads page](https://github.com/inspircd/inspircd/releases/latest).
